openhexa/sdk/pipelines/parameter.py

```diff
@@ -307,6 +308,34 @@ def to_connection(self, value: str) -> CustomConnection:
         return workspace.custom_connection(value)
 
 
+class DatasetType(ParameterType):
+    """Type class for dataset parameter."""
+
+    @property
+    def spec_type(self) -> str:
+        """Return a type string for the specs that are sent to the backend."""
+        return "dataset"
+
+    @property
+    def expected_type(self) -> type:
+        """Returns the python type expected for values."""
+        return DatasetType
```

Comment:
I think that this should be `Dataset` and not `DatasetType`. This comment is also valid for connection parameter types - `expected_type` should return the type expected for the **values** themselves. Could you check and fix it for all parameter types?